draft-duke-quic-load-balancers.md

```diff
@@ -59,270 +61,545 @@ author:
-
+ 
    This specification provides a method of coordination between QUIC
-   servers and layer 4 load balancers to support connection IDs that
+   servers and low-state load balancers to support connection IDs that
    encode routing information. It describes desirable properties of a
    solution, and then specifies a protocol that provides those
-   properties.
-
+   properties. This protocol supports multiple encoding schemes that
+   increase in complexity as they address paths between load balancer
+   and server with weaker trust dynamics.
+ 
 ## Terminology
-
+ 
    The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
    "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
    document are to be interpreted as described in RFC 2119 {{?RFC2119}}.
-
+ 
    In this document, these words will appear with that interpretation
    only when in ALL CAPS. Lower case uses of these words are not to be
    interpreted as carrying significance described in RFC 2119.
-
+ 
    In this document, "client" and "server" refer to the endpoints of a
    QUIC connection unless otherwise indicated. A "load balancer" is an
    intermediary for that connection that does not possess QUIC
    connection keys, but it may rewrite IP addresses or conduct other IP
-   or UDP processing. In most respects, the load balancer behaves as a
-   client in a QUIC-LB connection, but is always referred to as a "load
-   balancer" below to avoid confusion.
-
+   or UDP processing.
+   
+   Note that stateful load balancers that act as proxies, by
+   terminating a QUIC connection with the client and then retrieving
+   data from the server using QUIC or another protocol, are for all
+   intents and purposes a server with respect to this specification.
+
+   When discussing security threats to QUIC-LB, we distinguish between
+   “inside observers” and “outside observers.” The former lie on the
+   path between the load balancer and server, which often but not always
+   lies inside the server’s data center or cloud deployment. Outside
+   observers are on the path between the load balancer and client.
+   “Off-path” attackers, though not on any data path, may also be
+   “inside” or “outside” depending on whether not they have network
+   access to the server without intermediation by the load balancer
+   and/or other security devices.
+ 
 # Protocol Objectives
-
+ 
 ## Simplicity
-
+ 
    QUIC is intended to provide unlinkability across connection
    migration, but servers are under no obligation to provide connection
-   IDs to enable this. If the coordination scheme is too difficult to
-   implement, servers behind load balancers using connection IDs for
-   routing will use trivially linkable connection IDs. Clients will
-   therefore be forced choose between terminating the connection during
-   migration or remaining linkable.
-
+   IDs, or IDs that effectively prevent linkability. If the
```

Comment:
You state here that servers are `under no obligation to provide connection IDs, or IDs that effectively prevent linkability`. Do mean to say `to provide additional connections IDs, or`? It just sounds weird as is.